test/unit/format_capture_helper_test.rb

```diff
@@ -0,0 +1,87 @@
+require_relative "../test_helper"
+
+module MockCaptureHelper
+  def content_for(_name, content, _options, &_block)
+    content
+  end
+
+  def capture(&block)
+    block.call
+  end
+
+  def render(*_args, &block)
+    "<g>" + block.call + "</g>"
+  end
+end
+
+class MockClass
+  include MockCaptureHelper
+end
+
+module SmartAnswer
+  class FormatCaptureHelperTest < ActiveSupport::TestCase
+    def assert_captured_content(content, format, expected)
+      options = format.present? ? { format: format } : {}
+      args = [:name, expected, options]
+
+      MockClass.any_instance.expects(:content_for).with(*args)
+
+      @test_obj.content_for(:name, options) { content }
+    end
+
+    context "class is extended by helper" do
+      setup do
+        @test_obj = MockClass.new
+        @test_obj.extend(SmartAnswer::Helpers::FormatCaptureHelper)
+      end
+
+      should "call super with same arguments when no block given" do
+        MockClass.any_instance.expects(:content_for).with(:name, "content", { option: :option })
+
+        @test_obj.content_for(:name, "content", { option: :option }, &nil)
```

Comment:
I'm not sure this test is doing what it says it is. Since @test_obj is an instance of MockClass (and the only instance) it looks like this is calling a method on @test_obj and then asserting that exact same method was called.

Reply:
Yup this was some spooky behaviour - its setting up an expectation for the super method (i.e. MockClass#content_for), but here we are calling the (FormatCaptureHelper#content_for method). But with renaming `content_for` to `render_content_for` this should be clearer.